gbpservice/neutron/nsf/core/lb.py

```diff
@@ -0,0 +1,46 @@
+import os
+import sys
+
+
+class RoundRobin(object):
+
+    def __init__(self, workers):
+        self._workers = workers
+        self._rridx = 0
+        self._rrsize = len(self._workers)
+
+    def _rr(self):
+        item = self._workers[self._rridx]
+        self._rridx = (self._rridx + 1) % (self._rrsize)
+        return item
+
+    def get(self, rsrcid):
+        return self._rr()
+
+    def rem(self, rsrcid):
+        pass
+
+
+class StickyRoundRobin(object):
+
+    def __init__(self, workers):
+        self._workers = workers
+        self._assoc = {}
+        self._rridx = 0
+        self._rrsize = len(self._workers)
+
+    def _rr(self):
+        item = self._workers[self._rridx]
+        self._rridx = (self._rridx + 1) % (self._rrsize)
+        return item
+
+    def get(self, rsrcid):
+        if rsrcid in self._assoc.keys():
+            worker = self._assoc[rsrcid]
```

Comment:
"dict" is sort of hash right ? Here, in the logic we are trying to get the bound worker for the given resource identified by rsrc_id.